qlib/workflow/task/online.py

```diff
@@ -3,147 +3,140 @@
     def update_online_pred(self):
         """update all online model predictions to the latest day in Calendar"""
         mu = ModelUpdater(self.exp_name)
-        cnt = mu.update_all_pred(self.online_filter)
+        cnt = mu.update_all_pred(lambda rec: self.get_online_tag(rec) == OnlineManager.ONLINE_TAG)
         self.logger.info(f"Finish updating {cnt} online model predictions of {self.exp_name}.")
 
+    def after_day(self, *args, **kwargs):
+        self.prepare_signals(*args, **kwargs)
+        self.prepare_tasks(*args, **kwargs)
+        self.prepare_new_models(*args, **kwargs)
+        self.update_online_pred(*args, **kwargs)
+        self.reset_online_tag()
 
-class RollingOnlineManager(OnlineManager):
+
+class RollingOnlineManager(OnlineManagerR):
     def __init__(self, experiment_name: str, rolling_gen: RollingGen, task_pool) -> None:
         super().__init__(experiment_name)
         self.ta = TimeAdjuster()
         self.rg = rolling_gen
         self.tm = TaskManager(task_pool=task_pool)
-        self.logger = get_module_logger("RollingOnlineManager")
+        self.logger = get_module_logger(self.__class__.__name__)
 
-    def prepare_new_models(self):
-        """prepare(train) new models based on online model"""
-        latest_records = self.tc.list_latest_recorders(self.online_filter)  # if we need online_filter here?
-        max_test = self.tc.latest_time(latest_records)
+    def prepare_signals(self):
+        pass
+
+    def prepare_tasks(self):
+        latest_records, max_test = self.list_latest_recorders(lambda rec: self.get_online_tag(rec) == OnlineManager.ONLINE_TAG)
+        if max_test is None:
+            self.logger.warn(f"No latest_recorders.")
+            return
         calendar_latest = self.ta.last_date()
         if self.ta.cal_interval(calendar_latest, max_test[0]) > self.rg.step:
             old_tasks = []
             for rid, rec in latest_records.items():
-                task = self.tc.get_task(rec)
+                task = rec.load_object("task")
                 test_begin = task["dataset"]["kwargs"]["segments"]["test"][0]
                 # modify the test segment to generate new tasks
                 task["dataset"]["kwargs"]["segments"]["test"] = (test_begin, calendar_latest)
                 old_tasks.append(task)
             new_tasks = task_generator(old_tasks, self.rg)
-            self.tm.create_task(new_tasks)
-            run_task(task_train, self.tm.task_pool, experiment_name=self.exp_name)
-            self.logger.info(f"Finished prepare {len(new_tasks)} new models.")
-            return new_tasks
-        self.logger.info("No need to prepare any new models.")
-        return []
+            new_num = self.tm.create_task(new_tasks)
+            self.logger.info(f"Finished prepare {new_num} tasks.")
 
-    def prepare_signals(self):
-        # prepare the signals of today
-        pass
+    def prepare_new_models(self):
+        """prepare(train) new models based on online model"""
```

Comment:
I think we can create a class for training here.

Reply:
parallel training,   subclass  strange

Reply:
![image](https://user-images.githubusercontent.com/465606/113155232-62b33600-926b-11eb-9f7c-bc3c34b29a9a.png)